build-support/travis/travis.yml.mustache

```diff
@@ -125,38 +126,38 @@ base_linux_test_config: &base_linux_test_config
   before_script:
     - ./build-support/bin/get_ci_bootstrapped_pants_pex.sh ${BOOTSTRAPPED_PEX_BUCKET} ${BOOTSTRAPPED_PEX_KEY_PREFIX}.${BOOTSTRAPPED_PEX_KEY_SUFFIX}
 
-py2_linux_test_config: &py2_linux_test_config
-  <<: *py2_linux_config
+py27_linux_test_config: &py27_linux_test_config
```

Comment:
Shouldn't these now be using PY= exports to control 2.7, 3.6 and soon 3.7?

Reply:
This change would belong in `ci.sh` to avoid duplication in `.travis.yml`. My plan is in the followup PR is to introduce a `-7` flag that tells `ci.sh` to use Python 3.7, similar to how we pass `-2` now. There, for `-7`, I will use `PY=python3.7`.

For now, we don't need to make this change to use `PY=`:

- Python 2 case: when we pass `-2`, we instruct `ci.sh` to use `./pants` to bootstrap the PEX and to not override interpreter constraints. `./pants` defaults to using Python 2.7 to bootstrap, so no changes there. Subprocesses will already resolve to 2.7, as it's the minimum acceptable interpreter discovered in CI.
- Python 3 case: we default in `ci.sh` to using `./pants3`, which will use the minimum acceptable Python 3 interpreter (3.6 in CI) to bootstrap the PEX. For subprocesses, we constrain to `CPython>3.6,CPython<4`, which resolves to 3.6 as it's the minimum acceptable interpreter.

Although not necessary, I can add to `ci.sh` that the bootstrap script when using Py3 should be `PY=python3.6 ./pants3` and the `PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS` should be `CPython==3.6.*`. I did change the message we print to say `Using Python 3.6`, so makes sense to change the code to make sure it truly is, rather than relying on implicit assumptions.